spec/workspace-spec.js

```diff
@@ -116,6 +118,63 @@ describe('Workspace', () => {
         expect(atom.workspace.getTextEditors().length).toBe(0)
       })
     })
+
+    describe('where a dock contains an editor', () => {
+      afterEach(() => {
+        atom.workspace.getRightDock().paneContainer.destroy()
+      })
+
+      it('constructs the view with the same panes', () => {
```

Comment:
We have this file [`async-spec-helpers.js`](https://github.com/atom/atom/blob/master/spec/async-spec-helpers.js) that might help eliminate some of this `waitsFor` noise by enabling you to use `async`/`await`. Check out some other specs that use it for examples.